tests/test_init.py

```
@@ -216,23 +220,356 @@ def _create_client(**kwargs: Any) -> Any:
+        await init_mod.async_setup_entry(hass, entry)
+
+    coordinator.async_shutdown.assert_awaited_once()
+    client.async_close.assert_awaited_once()
+    assert entry.entry_id not in hass.data.get(init_mod.DOMAIN, {})
+    hass.config_entries.async_forward_entry_setups.assert_not_awaited()
+
+
+@pytest.mark.asyncio
+async def test_async_setup_entry_carp_platform_forward_failure_cleans_up(
+    monkeypatch: pytest.MonkeyPatch,
+    ph_hass: Any,
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `ph_hass`

(ANN401)